fineract-provider/src/main/java/org/apache/fineract/notification/service/NotificationMapperWritePlatformServiceImpl.java

```diff
@@ -38,4 +38,14 @@ public Long create(NotificationMapper notificationMapper) {
         this.notificationMapperRepository.saveAndFlush(notificationMapper);
         return notificationMapper.getId();
     }
+
+    @Override
+    public void markAllNotificationsForAUserAsRead(Long userId) {
+        notificationMapperRepository.markAllNotificationsForAUserAsRead(userId);
+    }
+
+    @Override
+    public void markASingleNotificationForAUserAsRead(Long userId, Long notificationId) {
```

Comment:
I'm missing some validations here around the fact that this way users can mark other users' notifications read. 
Whether a notification corresponds to a particular user shall be checked.

Reply:
In the update query there is an AND which ensures that it indeed for the logged in user.